prometheus/src/main/scala/filodb/prometheus/ast/Functions.scala

```diff
@@ -12,64 +13,100 @@ trait Functions extends Base with Operators with Vectors {
       RangeFunctionId.withNameLowercaseOnlyOption(name.toLowerCase).isEmpty &&
       FiloFunctionId.withNameLowercaseOnlyOption(name.toLowerCase).isEmpty &&
       MiscellaneousFunctionId.withNameLowercaseOnlyOption(name.toLowerCase).isEmpty &&
+      ScalarFunctionId.withNameInsensitiveOption(name.toLowerCase).isEmpty &&
       SortFunctionId.withNameLowercaseOnlyOption(name.toLowerCase).isEmpty) {
+
       throw new IllegalArgumentException(s"Invalid function name [$name]")
     }
 
+    /**
+      *
+      * @return true when function is scalar or time
+      */
+    def isScalarFunction(): Boolean = {
+       name.equalsIgnoreCase("scalar") ||
+        name.equalsIgnoreCase("time")
+    }
+
     // scalastyle:off
     def toPeriodicSeriesPlan(timeParams: TimeRangeParams): PeriodicSeriesPlan = {
-      val seriesParam = allParams.filter(_.isInstanceOf[Series]).head.asInstanceOf[Series]
-      val otherParams = allParams.filter(!_.equals(seriesParam)).map {
-        case num: ScalarExpression => num.toScalar
-        case s: InstantExpression  => s.realMetricName.replaceAll("^\"|\"$", "")
-        case _                     => throw new IllegalArgumentException("Parameters can be a string or number")
-      }
 
+      val vectorFn = VectorFunctionId.withNameInsensitiveOption(name)
       val instantFunctionIdOpt = InstantFunctionId.withNameInsensitiveOption(name)
       val filoFunctionIdOpt = FiloFunctionId.withNameInsensitiveOption(name)
       val miscellaneousFunctionIdOpt = MiscellaneousFunctionId.withNameInsensitiveOption(name)
+      val scalarFunctionIdOpt = ScalarFunctionId.withNameInsensitiveOption(name)
       val sortFunctionIdOpt = SortFunctionId.withNameInsensitiveOption(name)
 
-      if (instantFunctionIdOpt.isDefined) {
-        val instantFunctionId = instantFunctionIdOpt.get
-        val periodicSeriesPlan = seriesParam.asInstanceOf[PeriodicSeries].toPeriodicSeriesPlan(timeParams)
-
-        ApplyInstantFunction(periodicSeriesPlan, instantFunctionId, otherParams)
-      // Special FiloDB functions to extract things like chunk metadata
-      } else if (filoFunctionIdOpt.isDefined) {
-        // No lookback needed as we are looking at chunk metadata only, not raw samples
-        val rangeSelector = timeParamToSelector(timeParams, 0)
-        val (filters, columns) = seriesParam match {
-          case i: InstantExpression => (i.columnFilters :+ i.nameFilter, i.columns)
-          case r: RangeExpression   => (r.columnFilters :+ r.nameFilter, r.columns)
+      if (vectorFn.isDefined) {
+        allParams.head match {
+          case num: ScalarExpression => VectorPlan(ScalarFixedDoublePlan(num.toScalar, RangeParams(timeParams.start, timeParams.step, timeParams.end)))
+          case function: Function => val nestedPlan = function.toPeriodicSeriesPlan(timeParams)
```

Comment:
1) align the `=>`
2) Let's have the val start on the next line at indented level